smartrunner_driver/src/rosnode/smartrunner_node.cpp

```diff
@@ -16,26 +13,139 @@ extern "C"{
+    nh_.getParam("stereo_exposure_time", stereo_exposure_time_);
+    nh_.getParam("stereo_gain", stereo_gain_);
+    nh_.getParam("stereo_uniqueness", stereo_uniqueness_);
+ 
+    nh_.getParam("smartrunner_autotrigger_enable", smartrunner_autotrigger_enable_);
+    nh_.getParam("smartrunner_exposure_time", smartrunner_exposure_time_);
+    nh_.getParam("smartrunner_use_manual_exposure_time", smartrunner_use_manual_exposure_time_);
+    nh_.getParam("smartrunner_flash_time", smartrunner_flash_time_);
+    nh_.getParam("smartrunner_object_contrast", smartrunner_object_contrast_);
+    nh_.getParam("smartrunner_roi_min_x", smartrunner_roi_min_x_);
+    nh_.getParam("smartrunner_roi_max_x", smartrunner_roi_max_x_);
+    nh_.getParam("smartrunner_roi_min_z", smartrunner_roi_min_z_);
+    nh_.getParam("smartrunner_roi_max_z", smartrunner_roi_max_z_);
+    nh_.getParam("smartrunner_image_transfer_active", smartrunner_image_transfer_active_);
+ 
+    printf("ip: %s\n", device_ip_.c_str());
     printf("message type: %s\n", message_type_.c_str());
     printf("id: %s\n", frame_id_.c_str());
+    
+    printf("tof_trigger_source: %s\n", tof_trigger_source_.c_str());
+    printf("tof_auto_trigger_rate: %d\n", tof_auto_trigger_rate_);
+    printf("tof_trigger_enable: %d\n", tof_trigger_enable_);
+    printf("tof_exposure_time: %d\n", tof_exposure_time_);
+    printf("tof_range_mode: %d\n", tof_range_mode_);
+    printf("tof_output_mode: %s\n", tof_output_mode_.c_str());
 
-    if( scanner_ip_ == "" )
+    printf("stereo_trigger_source: %s\n", stereo_trigger_source_.c_str());
+    printf("stereo_auto_trigger_rate: %d\n", stereo_auto_trigger_rate_);
+    printf("stereo_trigger_enable: %d\n", stereo_trigger_enable_);
+    printf("stereo_exposure_time: %d\n", stereo_exposure_time_);
+    printf("stereo_gain: %d\n", stereo_gain_);
+    printf("stereo_uniqueness: %d\n", stereo_uniqueness_); 
+    printf("stereo_output_mode: %s\n", stereo_output_mode_.c_str());
+
+    printf("smartrunner_trigger_enable: %d\n", smartrunner_autotrigger_enable_);
+    printf("smartrunner_exposure_time_: %d\n", smartrunner_exposure_time_);
+    printf("smartrunner_use_manual_exposure_time_: %d\n", smartrunner_use_manual_exposure_time_);
+    printf("smartrunner_flash_time_: %d\n", smartrunner_flash_time_);
+    printf("smartrunner_object_contrast_: %d\n", smartrunner_object_contrast_);
+    printf("smartrunner_roi_min_x_: %d\n", smartrunner_roi_min_x_);
+    printf("smartrunner_roi_max_x_: %d\n", smartrunner_roi_max_x_);
+    printf("smartrunner_roi_min_z_: %d\n", smartrunner_roi_min_z_);
+    printf("smartrunner_roi_max_z_: %d\n", smartrunner_roi_max_z_);
+    printf("smartrunner_image_transfer_active_: %d\n", smartrunner_image_transfer_active_);
+ 
+    if( device_ip_ == "" )
     {
         printf("IP of scanner not set!\n");
         return;
     }
     else
     {
-        IP = scanner_ip_.c_str();
+        IP = device_ip_.c_str();
```

Comment:
c_str() returns only a temporary pointer. It must not be stored. (Make IP a std::string or copy the string if char buffer is needed)